zulipterminal/ui_tools/buttons.py

```diff
@@ -347,6 +359,36 @@ def _parse_narrow_link(cls, link: str) -> ParsedNarrowLink:
                 parsed_link['stream_id'] = stream_id
             return parsed_link
 
+        if (len_fragments == 5 and fragments[1] == 'stream'
+                and fragments[3] == 'near'):
+            stream_id, stream_name = cls._decode_stream_data(fragments[2])
+            message_id = cls._decode_message_id(fragments[4])
+            parsed_link = {'narrow': 'stream:near'}
+            if stream_name:
+                parsed_link['stream_name'] = stream_name
+            if stream_id:
+                parsed_link['stream_id'] = stream_id
+            if message_id:
+                parsed_link['message_id'] = message_id
```

Comment:
Rather than conditionally adding each of these, perhaps add them unconditionally (generate the full dict each time) and handle the validation slightly differently?